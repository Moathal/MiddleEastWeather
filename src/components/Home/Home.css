.home-Header {
  height: 20%;
  padding: 0;
  display: flex;
  align-items: center;
  width: calc(100% - 6px);
  justify-content: space-evenly;
  gap: 3px;
  box-shadow: 0 0 6px rgb(0, 29, 56);
  margin-bottom: 12px;
}

.countries {
  height: 25vh;
}

.countriesContainer {
  display: grid;
  grid-template-columns: 1fr 1fr;
  justify-content: center;
}

.search {
  width: 120px;
  border-radius: 6px;
  border: none;
  font-size: 12px;
  padding: 6px;
}
