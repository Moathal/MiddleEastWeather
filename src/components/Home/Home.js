import React, { useState } from 'react';
import map from '../../assets/Untitled-design.png';
import countries from '../countries';
import Country from '../Country/Country';
import './Home.css';

function Home() {
  const [value, setValue] = useState('');
  return (
    <>
      <div className="home-Header">
        <img className="countries" src={map} alt="middle east map" />
        <input onChange={(e) => setValue(e.target.value)} className="search" type="text" value={value} placeholder="Search a country.." />
      </div>
      <div className="countriesContainer">
        {countries.filter((country) => {
          if (value === '') return country;
          const searched = value.toLowerCase();
          if (country.name.toLowerCase().includes(searched)) return country;
          return null;
        })
          .map((country) => (
            <Country className="country" key={country.id} name={country.name} id={country.id} img={country.img} />
          ))}
      </div>
    </>
  );
}

export default Home;
