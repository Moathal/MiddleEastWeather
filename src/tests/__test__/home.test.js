import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Home from '../../components/Home/Home';

it('renders Header correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Home />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
