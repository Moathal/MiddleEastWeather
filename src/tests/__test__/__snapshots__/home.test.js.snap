// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
Array [
  <div
    className="home-Header"
  >
    <img
      alt="middle east map"
      className="countries"
      src="Untitled-design.png"
    />
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search a country.."
      type="text"
      value=""
    />
  </div>,
  <div
    className="countriesContainer"
  >
    <div>
      <a
        className="countryContainer"
        href="/details/Amman,jo"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Amman,jo.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Jordan
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Ankara,tr"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Ankara,tr.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Turkey
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Baghdad,iq"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Baghdad,iq.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Iraq
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Beirut,leb"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Beirut,leb.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Lebanon
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Cairo,eg"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Cairo,eg.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Egypt
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Damascus,sy"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Damascus,sy.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Syria
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Doha,qa"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Doha,qa.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Qatar
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Dubai,ae"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Dubai,ae.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            UAE
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Jerusalem,il"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Jerusalem,il.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Palestine
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Kuwait,kw"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Kuwait,kw.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Kuwait
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Manama,bh"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Manama,bh.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Bahrain
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Muscat,om"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Muscat,om.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Oman
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Riad,sa"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Riad,sa.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            KSA
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Sanaa,ye"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Sanaa,ye.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Yemen
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        className="countryContainer"
        href="/details/Tehran,ir"
        onClick={[Function]}
      >
        <img
          alt="country's Map"
          className="left"
          src="Tehran,ir.png"
        />
        <div
          className="right"
        >
          <img
            alt="goto icon"
            className="goto"
            src="group55.png"
          />
          <p
            className="name"
          >
            Iran
          </p>
        </div>
      </a>
    </div>
  </div>,
]
`;
